WordPress/Classes/ViewRelated/Jetpack/Branding/Common/JetpackButton.swift

```diff
@@ -0,0 +1,128 @@
+import UIKit
+import SwiftUI
+
+/// A "Jetpack powered" button with two different styles (`badge` or    `banner`)
+class JetpackButton: UIButton {
+
+    enum ButtonStyle {
+        case badge
+        case banner
+    }
+
+    private let style: ButtonStyle
+
+    private lazy var imageBackgroundView: UIView = {
+        let view = UIView()
+        view.backgroundColor = imageBackgroundColor
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    init(style: ButtonStyle) {
+        self.style = style
+        super.init(frame: .zero)
+        configureButton()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("Storyboard instantiation not supported.")
+    }
+
+    private var buttonBackgroundColor: UIColor {
+        switch style {
+        case .badge:
+            return UIColor(light: .muriel(color: .jetpackGreen, .shade40),
+                           dark: .muriel(color: .jetpackGreen, .shade90))
+        case .banner:
+            return .clear
+        }
+    }
+
+    private var buttonTintColor: UIColor {
+        switch style {
+        case .badge:
+            return UIColor(light: .white,
+                           dark: .muriel(color: .jetpackGreen, .shade40))
+        case .banner:
+            return .muriel(color: .jetpackGreen, .shade40)
+        }
+    }
+
+    private var buttonTitleColor: UIColor {
+        switch style {
+        case .badge:
+            return .white
+        case .banner:
+            return UIColor(light: .black, dark: .white)
+        }
+    }
+
+    private var imageBackgroundColor: UIColor {
+        switch style {
+        case .badge:
+            return UIColor(light: .muriel(color: .jetpackGreen, .shade40),
+                           dark: .white)
+        case .banner:
+            return .white
+        }
+    }
+
+    private func configureButton() {
+        // TODO: Remove this when the modal presentation is added
+        isUserInteractionEnabled = false
+        setTitle(Appearance.title, for: .normal)
+        tintColor = buttonTintColor
+        backgroundColor = buttonBackgroundColor
+        setTitleColor(buttonTitleColor, for: .normal)
+        titleLabel?.font = Appearance.titleFont
+        titleLabel?.adjustsFontForContentSizeCategory = true
+        titleLabel?.minimumScaleFactor = Appearance.minimumScaleFactor
+        titleLabel?.adjustsFontSizeToFitWidth = true
+        setImage(.gridicon(.plans), for: .normal)
+        contentVerticalAlignment = .fill
+        contentHorizontalAlignment = .center
```

Comment:
It looks like this [may be a default](https://developer.apple.com/documentation/uikit/uicontrol/1618213-contenthorizontalalignment)?